engine/crates/engine-v2/src/response/mod.rs

```diff
@@ -23,6 +25,20 @@ pub enum Response {
 // Our internal error struct shouldn't be accessible. It'll also need some context like
 // ResponseKeys to even just present paths correctly.
 pub struct Error<'a>(&'a GraphqlError);
+impl<'a> Error<'a> {
+    pub fn message(&self) -> &String {
```

Comment:
Nitpicking a bit, but any reason for this not to be `&str`?

```suggestion
    pub fn message(&self) -> &str {
```